frontend/src/v5/services/realtime/realtime.service.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ *  Copyright (C) 2022 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { io, Socket } from 'socket.io-client';
+
+interface IRoomType {
+	teamspace?: string;
+	model?: string;
+	project?: string;
+	notifications?: boolean
+}
+
+let socket:Socket = null;
+const roomsJoined:Record<string, number> = {};
+
+const roomTypeToId = (roomType:IRoomType) => (roomType.notifications ? 'notifications'
+	: JSON.stringify(roomType, Object.keys(roomType).sort()));
+
+const idToRoomType = (id: string): IRoomType => {
+	if (id === 'notifications') return { notifications: true };
+	return JSON.parse(id);
+};
+
+interface IChatConfig {
+	host: string;
+	path: string;
+	reconnectionAttempts: number
+}
+
+export const initializeSocket = ({ host, path, reconnectionAttempts }: IChatConfig) => {
+	socket = io(host, {
+		path,
+		reconnectionAttempts,
+		transports: ['websocket'],
+		reconnection: true,
+		reconnectionDelay: 500,
+	});
+
+	// eslint-disable-next-line @typescript-eslint/no-use-before-define
+	socket.on('connect', reJoinRooms);
```

Comment:
It doesn't look like `reJoinRooms` function calls this function , so can't we just move the `reJoinRooms` declaration above this?

Reply:
Yes, I just didn't liked that in that place is not grouped in my own mental map of logical blocks (all the join functions together)

Reply:
in that case maybe move this function down? or would that be odd too?